xlibre_ptrtypes.h has conflicting types for WindowPtr in 25.2.x on macOS

### Select the version

25.1.X

### Describe your issue

This commit has broken macOS build due to conflict with system headers: https://github.com/X11Libre/xserver/commit/976e6cdd1389857959e0c3b32605e09639b470db

### Steps to reproduce

Run the build

### What did you expect?

_No response_

### Additional Information

```
:info:build [2/122] /opt/local/bin/gcc-mp-16 -Ihw/xquartz/GL/libCGLCore.a.p -Ihw/xquartz/GL -I../xserver-25.2.2/hw/xquartz/GL -I. -I../xserver-25.2.2 -IXext -I../xserver-25.2.2/Xext -IXext/dri2 -I../xserver-25.2.2/Xext/dri2 -I../xserver-25.2.2/exa -Ifb -I../xserver-25.2.2/fb -I../xserver-25.2.2/glamor -Imi -I../xserver-25.2.2/mi -Imiext/damage -I../xserver-25.2.2/miext/damage -Imiext/shadow -I../xserver-25.2.2/miext/shadow -Imiext/sync -I../xserver-25.2.2/miext/sync -Idix -I../xserver-25.2.2/dix -Iinclude -I../xserver-25.2.2/include -Ihw/xquartz -I../xserver-25.2.2/hw/xquartz -Ihw/xquartz/xpr -I../xserver-25.2.2/hw/xquartz/xpr -I/opt/local/include/pixman-1 -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -std=gnu99 -O2 -g -fno-strict-aliasing -fvisibility=hidden -fno-common -Wchar-subscripts -Wchar-subscripts -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wbad-function-cast -Wold-style-definition -Wunused -Wuninitialized -Wshadow -Wmissing-noreturn -Wmissing-format-attribute -Wredundant-decls -Werror=implicit -Werror=nonnull -Werror=init-self -Werror=main -Werror=missing-braces -Werror=sequence-point -Werror=return-type -Werror=trigraphs -Werror=array-bounds -Werror=write-strings -Werror=address -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Woverride-init -Wshift-negative-value -Wvla -Wincompatible-pointer-types -pipe -Os -Wno-error=array-bounds= -Wno-error=implicit-function-declaration -Wno-error=int-conversion -Wno-error=incompatible-pointer-types -arch ppc -MD -MQ hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o -MF hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o.d -o hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o -c ../xserver-25.2.2/hw/xquartz/GL/capabilities.c
:info:build FAILED: [code=1] hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o 
:info:build /opt/local/bin/gcc-mp-16 -Ihw/xquartz/GL/libCGLCore.a.p -Ihw/xquartz/GL -I../xserver-25.2.2/hw/xquartz/GL -I. -I../xserver-25.2.2 -IXext -I../xserver-25.2.2/Xext -IXext/dri2 -I../xserver-25.2.2/Xext/dri2 -I../xserver-25.2.2/exa -Ifb -I../xserver-25.2.2/fb -I../xserver-25.2.2/glamor -Imi -I../xserver-25.2.2/mi -Imiext/damage -I../xserver-25.2.2/miext/damage -Imiext/shadow -I../xserver-25.2.2/miext/shadow -Imiext/sync -I../xserver-25.2.2/miext/sync -Idix -I../xserver-25.2.2/dix -Iinclude -I../xserver-25.2.2/include -Ihw/xquartz -I../xserver-25.2.2/hw/xquartz -Ihw/xquartz/xpr -I../xserver-25.2.2/hw/xquartz/xpr -I/opt/local/include/pixman-1 -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -std=gnu99 -O2 -g -fno-strict-aliasing -fvisibility=hidden -fno-common -Wchar-subscripts -Wchar-subscripts -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wbad-function-cast -Wold-style-definition -Wunused -Wuninitialized -Wshadow -Wmissing-noreturn -Wmissing-format-attribute -Wredundant-decls -Werror=implicit -Werror=nonnull -Werror=init-self -Werror=main -Werror=missing-braces -Werror=sequence-point -Werror=return-type -Werror=trigraphs -Werror=array-bounds -Werror=write-strings -Werror=address -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Woverride-init -Wshift-negative-value -Wvla -Wincompatible-pointer-types -pipe -Os -Wno-error=array-bounds= -Wno-error=implicit-function-declaration -Wno-error=int-conversion -Wno-error=incompatible-pointer-types -arch ppc -MD -MQ hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o -MF hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o.d -o hw/xquartz/GL/libCGLCore.a.p/capabilities.c.o -c ../xserver-25.2.2/hw/xquartz/GL/capabilities.c
:info:build In file included from ../xserver-25.2.2/include/os.h:60,
:info:build                  from ../xserver-25.2.2/hw/xquartz/GL/capabilities.c:41:
:info:build ../xserver-25.2.2/include/xlibre_ptrtypes.h:25:25: error: conflicting types for 'WindowPtr'; have 'struct _Window *'
:info:build    25 | typedef struct _Window *WindowPtr;
:info:build       |                         ^~~~~~~~~
:info:build In file included from /System/Library/Frameworks/ApplicationServices.framework/Frameworks/QD.framework/Headers/QD.h:41,
:info:build                  from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:30,
:info:build                  from ../xserver-25.2.2/hw/xquartz/GL/capabilities.c:35:
:info:build /System/Library/Frameworks/ApplicationServices.framework/Frameworks/QD.framework/Headers/QuickdrawTypes.h:1098:41: note: previous declaration of 'WindowPtr' with type 'WindowPtr' {aka 'struct OpaqueWindowPtr *'}
:info:build  1098 | typedef struct OpaqueWindowPtr*         WindowPtr;
:info:build       |                                         ^~~~~~~~~
:info:build ninja: build stopped: subcommand failed.
```

### Extra fields

- [x] I have checked the existing [issues](https://github.com/X11Libre/xserver/issues)
- [x] I have read the [Contributing Guidelines](https://github.com/X11Libre/xserver/blob/master/CONTRIBUTING.md)
- [x] I'd like to work on this issue

--- a/hw/xquartz/GL/capabilities.c
+++ b/hw/xquartz/GL/capabilities.c
@@ -26,8 +26,13 @@
 #include <stdlib.h>
 #include <assert.h>
 
-#define Cursor Mac_Cursor
-#define BOOL   Mac_BOOL
+/* QuickDraw in ApplicationServices conflicts with the basic X server headers
+ * (e.g. WindowPtr, now pulled in transitively via os.h -> xlibre_ptrtypes.h).
+ * Rename the QuickDraw symbols out of the way, matching sanitizedCarbon.h. */
+#define Cursor    Mac_Cursor
+#define BOOL      Mac_BOOL
+#define WindowPtr QD_WindowPtr
+#define Picture   QD_Picture
 #include <OpenGL/OpenGL.h>
 #include <OpenGL/gl.h>
 #include <OpenGL/glu.h>
@@ -35,6 +40,8 @@
 #include <ApplicationServices/ApplicationServices.h>
 #undef Cursor
 #undef BOOL
+#undef WindowPtr
+#undef Picture
 
 #include "capabilities.h"
 

